src/libcoretest/num/mod.rs

```diff
@@ -177,4 +177,61 @@ mod tests {
     test_impl_from! { test_u16i32, u16, i32 }
     test_impl_from! { test_u16i64, u16, i64 }
     test_impl_from! { test_u32i64, u32, i64 }
+
+    // Signed -> Float
+    test_impl_from! { test_i8f32, i8, f32 }
```

Comment:
@cuviper 
>Say log<F: Into<f64>>(self, base: F) -> f64, where you want the flexibility of float but the base is often integral. 

In this example `F == u64` is good too, because precise conversion is not required, so `Into` would probably be too strict.